package.nix

```
@@ -60,12 +63,26 @@ stdenv.mkDerivation {
     "wayland"
   ];
   buildPhase = ":";
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Binary Ninja";
+      exec = "binaryninja";
+      icon = "logo";
+      desktopName = "Binary Ninja";
+      comment = "Binary Ninja is an interactive decompiler, disassembler, debugger, and binary analysis platform built by reverse engineers, for reverse engineers";
+      categories = [ "Development" ];
+    })
+  ];
+
   installPhase = ''
     runHook preInstall
 
     mkdir -p $out/bin
     mkdir -p $out/opt
+    mkdir -p $out/share/pixmaps
     cp -r * $out/opt
+    cp $out/opt/docs/img/logo.png $out/share/pixmaps/
```

For license reasons it'd be preferable, IMO, to not include the actual logo in this repo. However, that's OK: the `/docs/` folder in the Binary Ninja zip files is just the online documentation. This same image is also available on the online docs site, so we can do this:
```suggestion
    cp ${
      fetchurl {
        url = "https://docs.binary.ninja/img/logo.png";
        hash = "sha256-TzGAAefTknnOBj70IHe64D6VwRKqIDpL4+o9kTw0Mn4=";
      }
    } $out/share/pixmaps/binaryninja.png
```

---

I was hoping such a solution would exists, I just couldn't figure that out with my nix knowledge. That looks perfect